### PR TITLE
Added action for Distributor

### PR DESCRIPTION
**Pull request by sschlein**
I've created a new action for uploads to https://wifidistribution.com and also added docs and a link to a tutorial for this action.

Issue: https://github.com/fastlane/fastlane/issues/4349

### CODE REVIEW

**Comment by sschlein** [comment]
I've added the issue for further discussions.

File: fastlane/docs/Actions.md [hunk 1]
```diff
@@ -1167,6 +1167,20 @@ testfairy(
...
 
+### [DISTRIBUTOR](https://wifidistribution.com)
+
+Upload your iOS build to [DISTRIBUTOR](https://wifidistribution.com)
+
+```ruby
+distributor(
+  api_token: "...",
+  app_id: "...",
+  ipa: "./build.ipa"
```
Comment (KrauseFx) [inline fastlane/docs/Actions.md]: Let's remove the `ipa` block here, as the user gets this for free when using `gym` 👍

**Comment by KrauseFx** [comment]
Thanks for your contribution @sschlein. I added a comment before it's ready to be merged. Could you also squash the commits to one? 👍

**Comment by KrauseFx** [comment]
Hey @sschlein, could you remove the `ipa` line in the PR and make sure the build succeeds? It seems like there is one code style issue, which you can correct using `rubocop -a`

**Comment by sschlein** [comment]
Hi @KrauseFx, I'm still working on the PR and just tested the commit squashing yesterday. I'll mark the `ipa` line as optional in the docs as you might still need it if you don't use `gym`. I'll also add an additional line to override the API URL because on dedicated or on-premise solutions of DISTRIBUTOR you might want to override this. 

I'll let you know if I'm finished (should be ready this weekend) and the PR has the quality we want. I didn't think of the github mails for my commits you got yesterday, sorry :-)

**Comment by sschlein** [comment]
Any idea why the test is failing? Seems not to be related to my changes.

**Comment by KrauseFx** [comment]
@sschlein Sorry about that, this happens when commits are squashed. Could you rebase from master? That usually fixes the problem

**Comment by KrauseFx** [comment]
Hey @sschlein, we just launched the fastlane plugin system, that allows you to build your own plugins without having to merge it into the main code base.

Get started here: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md#create-your-own-plugin
You can basically copy over the action you already built after running `fastlane new_plugin` 🚀 

Please let us know how that works for you and if there is any way we can help 👍

**Pull request closed without merging**